src/operator/slice_channel-inl.h

```diff
@@ -62,11 +67,63 @@ struct SliceChannelParam : public dmlc::Parameter<SliceChannelParam> {
   }
 };  // struct SliceChannelParam
 
+struct SplitKernel {
+  template<typename DType>
+  static MSHADOW_XINLINE void Map(size_t i,
+                                  const DType *in_data, DType** out_data, const size_t* indices,
+                                  size_t num_sections, size_t axis_size, size_t trailing_size) {
+    // printf("%lu\n", i);
+    size_t idx = i / trailing_size % axis_size;
+    size_t target = 0;
+    for (size_t section = 0; section < num_sections; target = section++) {
+      if (indices[section] > idx) {
```

Comment:
can we do preincrement (++section) as standard C++ idiom and assign target somewhere less ofuscated?